libs/feature/record/src/lib/mapper/elasticsearch.field.mapper.ts

```diff
@@ -98,6 +101,16 @@ export class ElasticsearchFieldMapper {
         selectField<SourceWithUnknownProps[]>(source, 'link')
       ).map(mapLink),
     }),
+    linkProtocol: (output, source) => {
+      const protocols = getAsArray(
+        selectField<SourceWithUnknownProps[]>(source, 'linkProtocol')
+      )
+      return {
+        ...output,
+        hasDownloads: this.linkHelper.hasDownloadProtocols(protocols),
+        hasMaps: this.linkHelper.hasMapApiProtocols(protocols),
```

Comment:
First, I didn't want to do it in the mapper because of the cyclic dependencies, also because i had the same concerns which you mentioned.
For my case (summary), doing the classification elsewhere (eg in the RecordPreview) means that we would have no consistency if we use the `state.results.records` elsewhere, and I think this information (download/view) could be useful to have it in the state. So I am in favor to do it in the mapper.

For the full record, everything, the request, the state, the links, are related to the record module behavior, so it could make sense to keep it there in the `mdview.facade`.
We could put it in the mapper it's true (also that now the mapper has been moved to the record module), but for now, the full mapping is only done for the mdview, so in the end, everything is coded all together it's ok for now.
If we request a full view for something else, it's not obvious that we will need the link classification but the future will tell.

My feeling is that we can leave it this way for now, we are not breaking the logic as we are talking about 2 different states.

Reply:
Ok that makes sense, I guess we'll see what the future requirements will be about and decide from that.